src/devices/bus/heathzenith/h8/cpu8080.cpp

```diff
@@ -0,0 +1,344 @@
+
+  H1-H3     Open      No       Tie Ready-in from IC212 Generator to CPU Wait line
+  H2-H3     Close     No       Tie Ready-in from IC212 Generator to CPU Ready line
+
+  J1-J2     Close     No       CPU HLDA out through flip-flop, use /Q output to E jumper
+  J2-J3     Open      No       CPU HLDA out through flip-flop, use Q output to E jumper
+
+  K1-K2     Close     No       Affects Data-in/out buffers enabled signals based on /MEMR & /IOR signals
+  K1-K3     Open      No       Affects Data-in/out buffers enabled signals based on /MEMR & /IOR signals
+
+  L1-L2     Close     No       Tie CPU /INTA to interrupt buffer
+  L2-L3     Open      No       Tie CPU /INTA to +12V
+
+  P1-P2     Open      No       A10 to pin 19 of IC204 ROM
+  P2-P3     Close     No       R jumper to pin 19 of IC204 ROM
+
+  R1-R2     Close     No       +12V selected
+  R2-R3     Open      No       S jumper selection selected
+
+  S1-S2     Open      No       +5V selected
+  S2-S3     Close     No       GND selected
+
+  T1-T2     Close     No       -5V to pin 21 of IC204 ROM
+  T2-T3     Open      No       +5V to pin 21 of IC204 ROM
+
+  X1-X2     Open      No       Allow /ROM_Disable from BH bus (pin 46) to disable ROM
+
+  Z1-Z2     Close     No       Tie /A10 to ROM decoder select line
+  Z2-Z3     Open      No       Tie +5V to ROM decoder select line
+
+  RDYIN     Close     No       Allow RDYIN from BH Bus (pin 20) to IC212 Generator
+
+****************************************************************************/
+
+#include "emu.h"
+#include "cpu8080.h"
+
+#include "bus/heathzenith/intr_cntrl/intr_cntrl.h"
+#include "cpu/i8085/i8085.h"
+
+
+namespace {
+
+class h_8_cpu_8080_device : public device_t, public device_h8bus_p2_card_interface
+{
+public:
+
+	h_8_cpu_8080_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock = 0);
+
+	virtual void int1_w(int state) override;
+	virtual void int2_w(int state) override;
+	virtual void int3_w(int state) override;
+	virtual void int4_w(int state) override;
+	virtual void int5_w(int state) override;
+	virtual void int6_w(int state) override;
+	virtual void int7_w(int state) override;
+
+	virtual void p201_reset_w(int state) override;
+	virtual void p201_int1_w(int state) override;
+	virtual void p201_int2_w(int state) override;
+
+protected:
+
+	virtual const tiny_rom_entry *device_rom_region() const override ATTR_COLD;
+	virtual ioport_constructor device_input_ports() const override ATTR_COLD;
+	virtual void device_start() override ATTR_COLD;
+	virtual void device_reset() override ATTR_COLD;
+	virtual void device_add_mconfig(machine_config &config) override ATTR_COLD;
+
+	void handle_int1();
+	void handle_int2();
+
+private:
+	void h8_status_callback(u8 data);
+	void h8_inte_callback(int state);
+
+	void mem_map(address_map &map) ATTR_COLD;
+	void io_map(address_map &map) ATTR_COLD;
+
+	void bus_mem_w(offs_t offset, u8 data) { h8bus().space(AS_PROGRAM).write_byte(offset, data); }
```

Comment:
It's preferred to use `memory_space::specific` instead of looking up the space each time.

Reply:
I think I found an example of that in altos586.cpp. Is that similar to how I should do it?

Reply:
Again, please do your own homework. You can search for any use of `memory_access::specific` in the code base and see how it works and then test your own implementation.

Reply:
Just to be clear, your initial comment said `memory_space::specific`, since I didn't find anything with `memory_space`, but did see a `memory_access::specific` used in the altos586 file, I wanted to make sure that was what you were referring to.

And related to asking for examples to look at, I do search for examples myself, but the issue that has come up more than once, is that I find something, but I'm not aware it is an outdated/bad approach, so once I'm done coding/testing/etc. and put up the PR, I get told it needs to totally change, and I've just wasted all that time. I also thought you may have worked on something similar and could identify the code using the current best practices without having to search for it. 

I'll work on addressing your comments.